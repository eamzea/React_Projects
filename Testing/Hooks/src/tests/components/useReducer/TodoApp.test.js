import "@testing-library/jest-dom";
import React from "react";
import { shallow, mount } from "enzyme";
import TodoApp from "../../../components/useReducer/TodoApp";
import { demoTodos } from "../../fixtures/demoTodos";
import { act } from "@testing-library/react";

describe("Pruebas en <TodoApp/>", () => {
  const wrapper = shallow(<TodoApp />);

  Storage.prototype.setItem = jest.fn(() => {});

  test("Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de agregar un TODO", () => {
    const wrapper = mount(<TodoApp />);

    act(() => {
      wrapper.find("TodoAdd").prop("handleAddTodo")(demoTodos[0]);
      wrapper.find("TodoAdd").prop("handleAddTodo")(demoTodos[1]);
    });

    expect(wrapper.find("p").text().trim()).toBe("To Do App (2)");
    expect(localStorage.setItem).toHaveBeenCalledTimes(2);
  });

  test("Debe borrar un TODO", () => {
    wrapper.find("TodoAdd").prop("handleAddTodo")(demoTodos[0]);
    expect(wrapper.find("p").text().trim()).toBe("To Do App (1)");

    wrapper.find("TodoList").prop("handleDelete")(demoTodos[0].id);
    expect(wrapper.find("p").text().trim()).toBe("To Do App (0)");
  });
});
