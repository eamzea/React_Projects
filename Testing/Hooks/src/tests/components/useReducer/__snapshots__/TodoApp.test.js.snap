// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoApp/> Debe de mostrarse correctamente 1`] = `
<div>
  <p
    className="h1"
  >
    To Do App (
    0
    )
  </p>
  <hr />
  <Row
    noGutters={false}
  >
    <Col
      xs={7}
    >
      <TodoList
        handleDelete={[Function]}
        handleToggle={[Function]}
        todos={Array []}
      />
    </Col>
    <Col
      xs={5}
    >
      <TodoAdd
        handleAddTodo={[Function]}
      />
    </Col>
  </Row>
</div>
`;
